admin/broadleaf-admin-module/src/main/resources/messages/UtilityMessages.properties

```diff
@@ -60,5 +60,6 @@ Validation_Failure=Entity failed validation for Duplication. Refresh the page an
 
 OfferCode_Duplication_Validation_Failure=Offer code with the same value already exist
 
-validateSelfLink=Do not add self link
-validationRecursiveRelationship=This linked upSale product cannot be used because it causes a recursive relationship with another product: {0}
+validateSelfLink=Cannot add itself
+upSaleProductValidationRecursiveRelationship=This linked upSale product cannot be used because it causes a recursive relationship with another product: {0}
+crossSaleProductValidationRecursiveRelationship=This linked cross sale product cannot be used because it causes a recursive relationship with another product: {0}
```

Comment:
Is it possible to use the same msg for all 3 prod validation (upscale,cross-sale & featured)? For example
```
validateProductSelfLink=Cannot add the product itself
validateProductRecursiveRelationship=This product cannot be used because it causes a recursive relationship with another product: {0}
```